tests/run/i12656.scala

```diff
@@ -0,0 +1,8 @@
+transparent inline def expectCompileError(
+  inline code: String,
+  expectedMsg: String
+) =
+  val errors = compiletime.testing.typeCheckErrors(code)
```

Comment:
We should add a similar test using `typeChecks` to make sure that one works and avoid regressions.